### PR TITLE
Recording the serial interval

### LINKED ISSUE #262: Keeping track of secondary infections
**Is your feature request related to a problem? Please describe.**
We wish to record the number of people each person infects over the course of the epidemic (each time they are re-infected) so that we can find the R_t value at each time step. The dataframes produced will be further used in the sampling package EpiOS.

**Describe the solution you'd like**
I will change the `Person` class to record a list of secondary infections each person has during the epidemic. I will also update the simulation to record and output a dataframe containing this information.

### PR DESCRIPTION
**Pull request by mghosh00**
## Summary

I have added the ability to record serial intervals of all infection events (which can be used to generate a distribution of serial intervals). The output of this is in a file named serial_intervals.csv.

## Checklist

- [x] All new functions have docstrings in the correct style
- [ ] I've verified the complete docs build locally without errors
- [x] I've maintained 100% coverage (please mention any 'no cover' annotations explicitly)
- [x] I've unit-tested all new methods directly

- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] I have read the **CONTRIBUTING** document.
- [x] My code follows the code style of this project.
- [ ] I have updated the wiki with new parameters/model functionality

## Closing issues

Closes #262

### CODE REVIEW

**Comment by KCGallagher** [comment]
@mghosh00 just following up on this so the repo is up to date for publication - how much of this is actually intended to go onto main? For example I see the (untested) triple sweep work here, which I assume was just a simple debugging test and so will not actually be merged into the main repo. If this is the case, it could be more appropriate to close the PR, and perhaps delete the branch if we do not now need this code.

**Comment by mghosh00** [comment]
Closes #273

File: pyEpiabm/pyEpiabm/sweep/place_sweep.py [hunk 1]
```diff
@@ -57,6 +57,18 @@ def __call__(self, time: float):
...
+                            # to be the exposure period of the infectee
+                            inf_to_exposed = \
+                                (time - infector.infection_start_times[-1])
+                            infectee.set_exposure_period(inf_to_exposed)
+                            # Finally, store the infector's latency period
+                            # within the infectee, which will be used in
+                            # calculating the generation_time
+                            (infectee
+                             .set_infector_latent_period(infector
+                                                         .latent_period))
```
Comment (KCGallagher) [inline pyEpiabm/pyEpiabm/sweep/place_sweep.py:71]: I don't love that this code is repeated in each sweep, and would rather it was handled in the queue sweep (or an infection sweep) to avoid repetition. If this code has to be in this file (i.e. because the infector is only known in these sweeps, as I can't remember), then perhaps consider a separate function, that might be a method of the abstract sweep class, which each individual sweep class can call.

File: python_examples/basic_infection_history_simulation/simulation_outputs/output.csv [hunk 2]
```diff
@@ -1,62 +1,62 @@
...

+12.0,99,0,0,0,0,0,0,0,1,0,0

+13.0,99,0,0,0,0,0,0,0,1,0,0

+14.0,99,0,0,0,0,0,0,0,1,0,0

+15.0,99,0,0,0,0,0,0,0,1,0,0

+16.0,99,0,0,0,0,0,0,0,1,0,0
```
Comment (KCGallagher) [inline python_examples/basic_infection_history_simulation/simulation_outputs/output.csv:18]: Did you mean to commit an updated version of this file? It seems to be a simulation where the pandemic doesn't take off and doesn't exemplify this software like the previous one. Might be best to revert these changes.

File: python_examples/basic_infection_history_simulation/triple_simulation_flow.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
+#
```
Comment (KCGallagher) [inline python_examples/basic_infection_history_simulation/triple_simulation_flow.py:1]: Do you mean to commit this file to main? I don't think we necessarily need it there, as I am not sure that it is useful to a public audience (and was just for internal debugging). If it has wider use then we should document this in the readme.

File: python_examples/basic_infection_history_simulation/triple_simulation_outputs/output.csv [hunk 4]
```diff
@@ -0,0 +1,22 @@
+time,InfectionStatus.Susceptible,InfectionStatus.Exposed,InfectionStatus.InfectASympt,InfectionStatus.InfectMild,InfectionStatus.InfectGP,InfectionStatus.InfectHosp,InfectionStatus.InfectICU,InfectionStatus.InfectICURecov,InfectionStatus.Recovered,InfectionStatus.Dead,InfectionStatus.Vaccinated
```
Comment (KCGallagher) [inline python_examples/basic_infection_history_simulation/triple_simulation_outputs/output.csv:1]: Ditto to previous point - what do these simulation results tell us? I thought they were just for our debugging, but if they have wider interest then we need to document that

File: pyEpiabm/pyEpiabm/sweep/place_sweep.py (hunk 1 above)
Comment (mghosh00) [reply pyEpiabm/pyEpiabm/sweep/place_sweep.py:71]: Ah yes, that makes sense, will work on that!

File: python_examples/basic_infection_history_simulation/simulation_outputs/output.csv (hunk 2 above)
Comment (mghosh00) [reply python_examples/basic_infection_history_simulation/simulation_outputs/output.csv:18]: Thanks for catching this, and no this was a mistake! (same for all the files you mentioned below)

File: python_examples/basic_infection_history_simulation/triple_simulation_flow.py (hunk 3 above)
Comment (mghosh00) [reply python_examples/basic_infection_history_simulation/triple_simulation_flow.py:1]: See prev comment (mistake)

File: python_examples/basic_infection_history_simulation/triple_simulation_outputs/output.csv (hunk 4 above)
Comment (mghosh00) [reply python_examples/basic_infection_history_simulation/triple_simulation_outputs/output.csv:1]: See prev comment (mistake)
